src/atomate2/ase/jobs.py

```diff
@@ -0,0 +1,308 @@
+                added to match the method signature of other makers.
+        """
+        raise NotImplementedError
+
+    @property
+    def calculator(self) -> Calculator:
+        """ASE calculator, method to be implemented in subclasses."""
+        raise NotImplementedError
+
+
+@dataclass
+class AseRelaxMaker(AseMaker):
+    """
+    Base Maker to calculate forces and stresses using any ASE calculator.
+
+    Should be subclassed to use a specific ASE. The user should
+    define `self.calculator` when subclassing.
+
+    Parameters
+    ----------
+    name : str
+        The job name.
+    relax_cell : bool = True
+        Whether to allow the cell shape/volume to change during relaxation.
+    fix_symmetry : bool = False
+        Whether to fix the symmetry during relaxation.
+        Refines the symmetry of the initial structure.
+    symprec : float | None = 1e-2
+        Tolerance for symmetry finding in case of fix_symmetry.
+    steps : int
+        Maximum number of ionic steps allowed during relaxation.
+    relax_kwargs : dict
+        Keyword arguments that will get passed to :obj:`AseRelaxer.relax`.
+    optimizer_kwargs : dict
+        Keyword arguments that will get passed to :obj:`AseRelaxer()`.
+    calculator_kwargs : dict
+        Keyword arguments that will get passed to the ASE calculator.
+    ionic_step_data : tuple[str,...] or None
+        Quantities to store in the TaskDocument ionic_steps.
+        Possible options are "struct_or_mol", "energy",
+        "forces", "stress", and "magmoms".
+        "structure" and "molecule" are aliases for "struct_or_mol".
+    store_trajectory : emmet .StoreTrajectoryOption = "no"
+        Whether to store trajectory information ("no") or complete trajectories
+        ("partial" or "full", which are identical).
+    tags : list[str] or None
+        A list of tags for the task.
+    """
+
+    name: str = "ASE relaxation"
+    relax_cell: bool = True
+    fix_symmetry: bool = False
+    symprec: float | None = 1e-2
+    steps: int = 500
+    relax_kwargs: dict = field(default_factory=dict)
+    optimizer_kwargs: dict = field(default_factory=dict)
+
+    @job(data=_ASE_DATA_OBJECTS)
+    def make(
+        self,
+        mol_or_struct: Molecule | Structure,
+        prev_dir: str | Path | None = None,
+    ) -> AseStructureTaskDoc | AseMoleculeTaskDoc:
+        """
+        Relax a structure or molecule using ASE as a job.
+
+        Parameters
+        ----------
+        mol_or_struct: .Molecule or .Structure
+            pymatgen molecule or structure
+        prev_dir : str or Path or None
+            A previous calculation directory to copy output files from. Unused, just
+                added to match the method signature of other makers.
+
+        Returns
+        -------
+        AseStructureTaskDoc or AseMoleculeTaskDoc
+        """
+        return AseTaskDoc.to_mol_or_struct_metadata_doc(
+            getattr(self.calculator, "name", self.calculator.__class__),
```

Comment:
i think this line will fail `pydantic` validation if a calculator should ever not have a name. `__class__` is not a string, we want the `__name__` attribute

https://github.com/materialsproject/atomate2/pull/940/files#diff-9eab185b405c88913d9a52dc90990b06235a6955a91c6c06ccbc899a4e3ec5b7R204-R207

Reply:
good catch!